packages/integration-tests/tests/social-session.test.ts

```diff
@@ -0,0 +1,121 @@
+  });
+
+  it('register with social', async () => {
+    const client = new MockClient();
+
+    await client.initSession();
+    assert(client.interactionCookie, new Error('Session not found'));
+
+    await expect(
+      signInWithSocial(
+        { state, connectorId: mockSocialConnectorId, redirectUri },
+        client.interactionCookie
+      )
+    ).resolves.toBeTruthy();
+
+    const response = await getAuthWithSocial(
+      { connectorId: mockSocialConnectorId, data: { state, redirectUri, code, userId } },
+      client.interactionCookie
+    ).catch((error: unknown) => error);
+
+    // User with social does not exist
+    expect(response instanceof HTTPError && response.response.statusCode === 422).toBe(true);
+
+    // Register with social
+    const { redirectTo } = await registerWithSocial(
+      mockSocialConnectorId,
+      client.interactionCookie
+    );
+
+    await client.processSession(redirectTo);
+
+    expect(client.isAuthenticated).toBeTruthy();
+  });
+
+  /* Note: As currently we can not prepare a social identities through admin api.
+   * The sign-in test case MUST run concurrently after the register test case
+   */
+  it('Sign-In with social', async () => {
+    const client = new MockClient();
+
+    await client.initSession();
+    assert(client.interactionCookie, new Error('Session not found'));
+
+    await expect(
+      signInWithSocial(
+        { state, connectorId: mockSocialConnectorId, redirectUri },
+        client.interactionCookie
+      )
+    ).resolves.toBeTruthy();
+
+    const { redirectTo } = await getAuthWithSocial(
+      { connectorId: mockSocialConnectorId, data: { state, redirectUri, code, userId } },
+      client.interactionCookie
+    );
+
+    await client.processSession(redirectTo);
+
+    expect(client.isAuthenticated).toBeTruthy();
+  });
+});
+
+describe('social bind account', () => {
+  const username = generateUsername();
+  const password = generatePassword();
+
+  beforeAll(async () => {
+    await createUserByAdmin(username, password);
+  });
+
+  it('bind new social account', async () => {
+    const client = new MockClient();
+
+    await client.initSession();
+    assert(client.interactionCookie, new Error('Session not found'));
+
+    await expect(
+      signInWithSocial(
+        { state, connectorId: mockSocialConnectorId, redirectUri },
+        client.interactionCookie
+      )
+    ).resolves.toBeTruthy();
+
+    const response = await getAuthWithSocial(
+      { connectorId: mockSocialConnectorId, data: { state, redirectUri, code } },
+      client.interactionCookie
+    ).catch((error: unknown) => error);
+
+    // User with social does not exist
+    expect(response instanceof HTTPError && response.response.statusCode === 422).toBe(true);
+
+    await signInWithUsernameAndPassword(username, password, client.interactionCookie);
+
+    await expect(
+      bindWithSocial(mockSocialConnectorId, client.interactionCookie)
+    ).resolves.not.toThrow();
+  });
```

Comment:
Maybe need to check if there are `identities` inside the user info data.

Reply:
should throw if not included?

Reply:
updated